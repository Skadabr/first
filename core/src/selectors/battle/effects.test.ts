import { state, State } from "../../__tests__/fixtures/state-fixture";
import {
  getEffects,
  getEffectsApplicableToUnit,
  getEffectsByImpact,
  getEffectsByUnitId,
  getEffectsByUserId,
  isEffectApplicableToUnit
} from "./effects";
import { EffectImpact, EffectTargetingScope } from "../../index";

describe("getEffects", () => {
  it("should return list of effects", () => {
    expect(getEffects(state)).toMatchSnapshot();
  });
});

describe("getEffectByUnitId", () => {
  it("should return effects of unit", () => {
    expect(getEffectsByUnitId(state, "pl_pawn2")).toMatchSnapshot();
  });
});

describe("getEffectByUserId", () => {
  it("should return effects of user", () => {
    expect(getEffectsByUserId(state, "player")).toMatchSnapshot();
  });
});

describe("getEffectByImpact", () => {
  it("should return effects with `Health` impact", () => {
    expect(getEffectsByImpact(state, EffectImpact.Health)).toMatchSnapshot();
  });
});

describe("isEffectApplicableToUnit", () => {
  describe("when we take effect which is targeting only other friendly minions", () => {
    let effect;

    beforeEach(function() {
      effect = getEffectsByUnitId(state, "pl_pawn2")[0];
    });

    it("should target OtherFriendlyMinions", function() {
      expect(effect.targetingScope).toEqual(
        EffectTargetingScope.OtherFriendlyMinions
      );
    });

    it("should not target on enemy minions", function() {
      const res = isEffectApplicableToUnit(state, effect, "e_pawn3");
      expect(res).toBe(false);
    });

    it("should target on other friendly minions", function() {
      const res = isEffectApplicableToUnit(state, effect, "pl_pawn1");
    });
  });
});

describe("getEffectsApplicableToUnit", () => {
  it("should return all effects which can be applicable to unit (impact on its behavior)", function() {
    const effs = getEffectsApplicableToUnit(state, "pl_pawn1");
    expect(effs).toMatchSnapshot();
  });
});

describe.skip("getUniqueListOfTradeEffectTypes", () => {});
