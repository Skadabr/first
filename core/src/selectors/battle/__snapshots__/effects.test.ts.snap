// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEffectByImpact should return effects with \`Health\` impact 1`] = `
Array [
  Object {
    "impact": "HEALTH",
    "ownerId": "pl_pawn2",
    "targetingScope": "OTHER_FRIENDLY_MINIONS",
    "type": "HEALTH",
    "value": 2,
  },
]
`;

exports[`getEffectByUnitId should return effects of unit 1`] = `
Array [
  Object {
    "impact": "HEALTH",
    "ownerId": "pl_pawn2",
    "targetingScope": "OTHER_FRIENDLY_MINIONS",
    "type": "HEALTH",
    "value": 2,
  },
]
`;

exports[`getEffectByUserId should return effects of user 1`] = `
Array [
  Object {
    "impact": "HEALTH",
    "ownerId": "pl_pawn2",
    "targetingScope": "OTHER_FRIENDLY_MINIONS",
    "type": "HEALTH",
    "value": 2,
  },
  Object {
    "impact": "ATTACK",
    "ownerId": "pl_pawn3",
    "targetingScope": "OTHER_FRIENDLY_MINIONS",
    "type": "ATTACK",
    "value": 1,
  },
]
`;

exports[`getEffects should return list of effects 1`] = `
Array [
  Object {
    "impact": "HEALTH",
    "ownerId": "pl_pawn2",
    "targetingScope": "OTHER_FRIENDLY_MINIONS",
    "type": "HEALTH",
    "value": 2,
  },
  Object {
    "impact": "ATTACK",
    "ownerId": "pl_pawn3",
    "targetingScope": "OTHER_FRIENDLY_MINIONS",
    "type": "ATTACK",
    "value": 1,
  },
]
`;

exports[`getEffectsApplicableToUnit should return all effects which can be applicable to unit (impact on its behavior) 1`] = `
Array [
  Object {
    "impact": "HEALTH",
    "ownerId": "pl_pawn2",
    "targetingScope": "OTHER_FRIENDLY_MINIONS",
    "type": "HEALTH",
    "value": 2,
  },
  Object {
    "impact": "ATTACK",
    "ownerId": "pl_pawn3",
    "targetingScope": "OTHER_FRIENDLY_MINIONS",
    "type": "ATTACK",
    "value": 1,
  },
]
`;
